File: Copilot/sistemaTipos/tipos.c
#include <stdio.h>
#include <stdlib.h>

double precio_final(double precio, const char *descuento_texto) {
	double descuento = strtod(descuento_texto, NULL);
	return precio - (precio * descuento / 100.0);
}

int main(void) {
	double precio_final_resultado = precio_final(50, "10");
	printf("%.2f\n", precio_final_resultado);
	return 0;
}

File: Copilot/sistemaTipos/tipos.rs
fn precio_final(precio: f64, descuento_texto: &str) -> f64 {
	let descuento: f64 = descuento_texto.parse().expect("descuento invalido");
	precio - (precio * descuento / 100.0)
}

fn main() {
	let precio_final = precio_final(50.0, "10");
	println!("{precio_final}");
}
